### PR TITLE
[SIG-3091] Change subcategory in the backoffice

### PR DESCRIPTION
**Pull request by aditudorache**
This PR adds the ability for for the authorized user to overrule the ML tool predictions. For this:
- a new `CategorySelect` and `CategorySelectRenderer` were introduced to handle the specific
- the `category` and `subcategory` string properties of the incident were merged in one object property that holds the selected category
- the `categoryPrediction` property is now of the same type as the `category` property of the incident
- when determining the classification prediction, the `subcategory` together with the `handling_message` is read from the api for compatibility frontoffice/backoffice. 
- the `handling_message` is not read anymore before creating the incident while is already present

### CODE REVIEW

File: src/models/categories/selectors.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import { createSelector } from 'reselect';
 
 import { initialState } from './reducer';
 
+export const getCategoryData = ({ id, name, slug, handling_message }) => ({ sub_category: id, name, slug, handling_message });
```
Comment (janjaap) [inline src/models/categories/selectors.js:6]: Why break the `select...` naming pattern here?

File: src/signals/incident-management/containers/LegacyIncidentSplitContainer/components/IncidentPart/index.test.js [hunk 2]
```diff
@@ -14,11 +14,6 @@ jest.mock('containers/App/selectors', () => ({
   ...jest.requireActual('containers/App/selectors'),
 }));
 
-const subCategories = categoriesFixture.results
-  .filter(modelSelectors.filterForSub)
-  // mapping subcategories to prevent a warning about non-unique keys rendered by the SelectInput element 🙄
-  .map(subCat => ({ ...subCat, key: subCat._links.self.href }));
-
```
Comment (janjaap) [inline src/signals/incident-management/containers/LegacyIncidentSplitContainer/components/IncidentPart/index.test.js:21]: Is this a change that is required for changing the subcategory in the incident form?

File: src/signals/incident/components/form/DescriptionInput/DescriptionInput.js [hunk 3]
```diff
@@ -24,7 +14,15 @@ const DescriptionInput = ({
...
   );
 
-  const handleBlur = useCallback(event => get(event, meta, parent), [meta, parent]);
+  const handleBlur = useCallback(event => {
+    const {
+      getClassification,
+      updateIncident,
+      incidentContainer: { usePredictions },
+    } = parent?.meta;
+    if (usePredictions && event.target.value) getClassification(event.target.value);
```
Comment (janjaap) [inline src/signals/incident/components/form/DescriptionInput/DescriptionInput.js:23]: Personally, I'd go for readable code and change this line in
```javascript
if (usePredictions && event.target.value) {
  getClassification(event.target.value);
}
```

File: src/signals/incident/containers/IncidentContainer/reducer.js [hunk 4]
```diff
@@ -33,29 +33,34 @@ export const initialState = fromJS({
...
   const seq = Seq(incident).filter((val, key) => !key.startsWith('extra'));
 
   return seq.toMap();
 };
 
+const getIncidentCategory = (state, category) => {
+  const prevCategory = state.get('incident').get('category')?.toJS();
+  const categoryPrediction = state.get('categoryPrediction')?.toJS();
+  const canChange = categoryPrediction === null || prevCategory === null || prevCategory?.slug === categoryPrediction?.slug;
+  return canChange ? category : prevCategory;
```
Comment (janjaap) [inline src/signals/incident/containers/IncidentContainer/reducer.js:61]: Checking the coverage, this branch isn't covered.

File: src/signals/incident/components/form/CategorySelect/CategorySelect.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      parent.meta.updateIncident({
+        category,
+        handling_message,
+      });
+    },
+    [parent, subcategories]
+  );
+
+  return (
+    <Select name={meta.name} value={`${handler().value.slug}`} onChange={handleChange} options={options || []} />
```
Comment (janjaap) [inline src/signals/incident/components/form/CategorySelect/CategorySelect.js:26]: `options || []` isn't covered by a test case

**Review by janjaap — COMMENTED** [review]
A couple of remarks.
In general, this PR is so big and contains changes all across the codebase, that it is hard to assess the impact of the changes without checking out the branch locally, clicking through the application's interface and running tests manually.

File: src/signals/incident-management/containers/LegacyIncidentSplitContainer/components/IncidentPart/index.test.js (hunk 2 above)
Comment (aditudorache) [reply src/signals/incident-management/containers/LegacyIncidentSplitContainer/components/IncidentPart/index.test.js:21]: no.

File: src/models/categories/selectors.js (hunk 1 above)
Comment (aditudorache) [reply src/models/categories/selectors.js:6]: This is not a selector, just a service utility, there is in this file `getHasParent` that uses the same naming pattern.

File: src/signals/incident-management/containers/LegacyIncidentSplitContainer/components/IncidentPart/index.test.js (hunk 2 above)
Comment (aditudorache) [reply src/signals/incident-management/containers/LegacyIncidentSplitContainer/components/IncidentPart/index.test.js:21]: no

File: src/signals/incident/components/form/DescriptionInput/DescriptionInput.js (hunk 3 above)
Comment (aditudorache) [reply src/signals/incident/components/form/DescriptionInput/DescriptionInput.js:23]: agree!

File: src/signals/incident/containers/IncidentContainer/reducer.js (hunk 4 above)
Comment (aditudorache) [reply src/signals/incident/containers/IncidentContainer/reducer.js:61]: Good catch! Thanks!

**Review by janjaap — APPROVED** [review]

**Pull request merged**